### PR TITLE
Fixing evm_reset method

### PR DESCRIPTION
**Pull request by asoto-iov**
Fix related with issue https://github.com/rsksmart/rskj/issues/1012 

## Description
When calling evm_reset method the Block Info where not cleared from index correctly, so old hashes are not being cleared from the index database.

## How Has This Been Tested?
Check issue linked above

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

* **Other information**:

### CODE REVIEW

**Comment by asoto-iov** [comment]
pipeline:run

File: rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java [hunk 1]
```diff
@@ -113,6 +114,33 @@ public void putBlocks(long blockNumber, List<IndexedBlockStore.BlockInfo> blocks
         index.put(blockNumber, blocks);
     }
 
+    @Override
+    public void removeBlock(long blockNumber, Keccak256 blockHash) {
+        if (!index.containsKey(blockNumber)) {
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java:119]: not sure we need this extra check.. it's basically two db queries - you already do same when checking `binfos` for `null` below

File: rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java [hunk 2]
```diff
@@ -113,6 +114,33 @@ public void putBlocks(long blockNumber, List<IndexedBlockStore.BlockInfo> blocks
...
+            return;
+        }
+
+        List<IndexedBlockStore.BlockInfo> binfos = index.get(blockNumber);
+
+        if (binfos == null) {
+            return;
+        }
+
+        List<IndexedBlockStore.BlockInfo> toremove = new ArrayList<>();
```
Comment (Vovchyk) [inline rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java:129]:
```suggestion
        List<IndexedBlockStore.BlockInfo> toRemove = new ArrayList<>();
```

**Review by Vovchyk — CHANGES_REQUESTED** [review]

File: rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java (hunk 1 above)
Comment (asoto-iov) [reply rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java:119]: that's true

**Review by Vovchyk — DISMISSED** [review]

File: rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java [hunk 3]
```diff
@@ -113,6 +114,29 @@ public void putBlocks(long blockNumber, List<IndexedBlockStore.BlockInfo> blocks
...
+        for (IndexedBlockStore.BlockInfo bInfo : blockInfoList) {
+            if (bInfo.getHash().equals(blockHash)) {
+                toRemove.add(bInfo);
+            }
+        }
+        blockInfoList.removeAll(toRemove);
+        if (blockInfoList.isEmpty()) {
+            index.remove(blockNumber);
+        } else {
+            index.put(blockNumber, blockInfoList);
```
Comment (illuque-iov) [inline rskj-core/src/main/java/co/rsk/db/MapDBBlocksIndex.java:136]: Maybe we can add a comment explaining why this is needed, so the next dev working here does not try to change it.

File: rskj-core/src/test/java/co/rsk/db/HashMapBlocksIndex.java [hunk 4]
```diff
@@ -67,4 +68,30 @@ public void flush() {
...
+
+        for (IndexedBlockStore.BlockInfo bInfo : blockInfoList) {
+            if (bInfo.getHash().equals(blockHash)) {
+                toRemove.add(bInfo);
+            }
+        }
+        blockInfoList.removeAll(toRemove);
+        if (blockInfoList.isEmpty()) {
+            index.remove(blockNumber);
+        }
```
Comment (illuque-iov) [inline rskj-core/src/test/java/co/rsk/db/HashMapBlocksIndex.java:94]: In line with my comment in `MapDBBlocksIndex.java`, we can comment why here we don't need to do the put.

**Review by illuque-iov — APPROVED** [review]

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by asoto-iov** [comment]
pipeline:run

**Comment by asoto-iov** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by asoto-iov** [comment]
pipeline:run

**Comment by asoto-iov** [comment]
pipeline:run

**Comment by asoto-iov** [comment]
pipeline:run

**Comment by asoto-iov** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Comment by Vovchyk** [comment]
pipeline:run

**Pull request merged**